contracts/oracle/src/lib.rs

```diff
@@ -165,16 +165,16 @@ impl Contract {
     }
 
     #[private]
-    #[handle_result]
     pub fn sbt_mint_callback(
         &mut self,
         external_id: &Vec<u8>,
-        #[callback_result] last_result: Result<TokenId, near_sdk::PromiseError>,
-    ) -> Result<TokenId, near_sdk::PromiseError> {
-        if last_result.is_err() {
-            self.used_identities.remove(&external_id);
+        #[callback_result] last_result: Result<TokenId, PromiseError>,
+    ) -> Option<TokenId> {
+        if last_result.is_ok() {
+            self.used_identities.insert(&external_id);
```

Comment:
this is wrong, it was already inserted. We need to remove (rollback) if mint in registry was not succesful.

Reply:
Done